lib/drivers/medtronic/medtronicSimulator.js

```diff
@@ -83,6 +83,17 @@ exports.make = function(config){
     events.push(e);
   }
 
+  function addDurationToTime(time, timezoneOffset, duration) {
```

Comment:
I would personally find it confusing to use it where it's not needed because it's obfuscating and will likely confuse future devs (or even future you 😜 ), but no, there isn't a reason not to...it's just vacuous.

If I were explaining the problem you were having before, I wouldn't say `Date.parse` was to blame. It was using `deviceTime`s _as if_ they were `time`s. `Date.parse` is always safe on `time` because `time` is a complete timezone-aware timestamp (in this case UTC) and so `Date.parse` just turns it from String to integer. `deviceTime` is **part of** a timezone-aware timestamp too, but in this case the entire timezone-aware timestamp is represented in pieces with its timezone (offset) information  "carried" externally (in `timezoneOffset`), so `Date.parse` on _just_ the `deviceTime` doesn't do the complete operation needed. Does that help?